flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClientTest.java

```diff
@@ -250,6 +251,41 @@ public void testCreateFlinkTaskManagerPod() throws Exception {
                 resultTaskManagerPod.getMetadata().getOwnerReferences().get(0).getUid());
     }
 
+    @Test
+    public void testCreateTwoTaskManagerPods() throws Exception {
```

Comment:
I suggest to introduce the `KubernetesClientTestBase#mockGetDeploymentWithError` to avoid exposing the internal state of `Fabric8FlinkKubeClient`.


```
    protected void mockGetDeploymentWithError(String deploymentId) {
        final String path =
                String.format(
                        "/apis/apps/v1/namespaces/%s/deployments/%s", NAMESPACE, deploymentId);
        server.expect()
                .get()
                .withPath(path)
                .andReturn(500, "Expected error")
                .always();
    }
```


After then, the test could be simplified like following. Please make sure the test failed before this PR and pass after this PR.
```
    @Test
    public void testCreateTwoTaskManagerPods() throws Exception {
        flinkKubeClient.createJobManagerComponent(this.kubernetesJobManagerSpecification);
        flinkKubeClient.createTaskManagerPod(buildTaskManagerPod("mock-task-manager-pod1")).get();
        mockGetDeploymentWithError(CLUSTER_ID);
        try {
            flinkKubeClient
                    .createTaskManagerPod(buildTaskManagerPod("mock-task-manager-pod2"))
                    .get();
        } catch (Throwable throwable) {
            fail("Should not get the JobManager deployment more than once");
        }
    }
```

Reply:
Thank you so much for your comments. I updated the code and PR description.